src/compute/sort/utf8.rs

```diff
@@ -9,9 +9,13 @@ pub(super) fn indices_sorted_unstable_by<I: Index, O: Offset>(
     options: &SortOptions,
     limit: Option<usize>,
 ) -> PrimitiveArray<I> {
-    let get = |idx| unsafe { array.value_unchecked(idx as usize) };
+    let values = unsafe {
+        (0..array.len())
+            .map(|idx| array.value_unchecked(idx as usize))
+            .collect::<Vec<_>>()
```

Comment:
won't this cause a large allocation?

Reply:
Yes, the quick-sort time complex os `O(n * log(n) )`,  means it'll take at least `O(n * log(n) )` times get from the buffer. I think it's worth to initialized it in a large vector.

I need to do some benches to verify Utf8 sort  compare with master.